### ISSUE TITLE
Hung process with 100% CPU usage on 4.1.6

### ISSUE THREAD

**Issue by andredasilvapinto** [lwouis/alt-tab-macos#410]
**Describe the bug**

Opening version 4.1.6 (latest) of the application results in 100% CPU usage + hung process.

OS: macOS 10.15.5

**Steps to reproduce the bug**

1. Try to open the application

**Comment by andredasilvapinto** [comment]
Everything works fine with 4.1.5

**Comment by lwouis** [comment]
Hi @andredasilvapinto! Someone else also got this issue in #401. It doesn't happen on my machine for some reason.

Could you please let me know if [this build](https://github.com/lwouis/alt-tab-macos/files/4902772/AltTab.app.zip) works for you? If it does, i'll push a new version right away.

**Comment by andredasilvapinto** [comment]
That build has the same issue on my computer (100% CPU usage and hung process).

**Comment by lwouis** [comment]
Ok I removed the cap, as it seemed to fix it for the other user. Could you please try this [new build](https://github.com/lwouis/alt-tab-macos/files/4903741/AltTab.app.zip)?

**Comment by andredasilvapinto** [comment]
yes, that works.

**Comment by ThatsJustCheesy** [comment]
I had this too and was about to report, but I'll try the new build instead!

If it's helpful, I did take a spindump of the hang ([AltTab Spindump.txt](https://github.com/lwouis/alt-tab-macos/files/4904525/AltTab.Spindump.txt)). I have also been hacking on another app that uses accessibility API, and have seen conflicts between accessibility apps in the past, so perhaps(?) that's a clue.

Thanks!

**Comment by lwouis** [comment]
>  I did take a spindump of the hang (AltTab Spindump.txt)

I was confused at first why the file was so insanely long. I think you spindump'ed all processes on your system. It must have taken a while to load, no?

> I have also been hacking on another app that uses accessibility API, and have seen conflicts between accessibility apps in the past, so perhaps(?) that's a clue.

No it's a deadlock situation. Data race or just recursive starvation at some point. Uncapping the concurrency, which is bad, seems to work, and it was uncapped for a long time so I just put it back that way. There is quite complex concurrency logic in this app at this point. It makes it hard to follow stuff sometimes, but the reward is buttery smooth responsiveness from the app when pressing shortcuts, even when there is activity around like apps launching and windows minimizing for example.

I finally passed CI after a few hours of fighting with cocoapods terribleness. A new release will be available shortly after CD finishes in a moment. Hopefully the issue is gone for everyone!

**Comment by ThatsJustCheesy** [comment]
> I was confused at first why the file was so insanely long. I think you spindump'ed all processes on your system.

Haha you're right, I just clicked AltTab in Activity Monitor and selected "Spindump", but I guess that's what it does (probably should have gone for "Sample Process"; not quite sure what I was thinking!)

Anyway, the new build fixed it for me! 👍

**Comment by lwouis** [comment]
@andredasilvapinto @ThatsJustCheesy is it back to working good for you on 4.2.0? Can I close this ticket?

**Comment by ThatsJustCheesy** [comment]
OK from me!

**Comment by lwouis** [comment]
Closing this ticket. @andredasilvapinto if you still experience the issue, please let me know here and I'll re-open the ticket 👍

**Comment by andredasilvapinto** [comment]
just tested 4.3.0. works fine. thanks!
